TechnicoBackend/Controllers/AuthController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using TechnicoBackend.Repositories;
+using TechnicoBackend.Models;
+using TechnicoBackend.DTOs;
+
+namespace TechnicoBackend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class AuthController : ControllerBase
+    {
+        private readonly UserRepository _userRepository;
```

Comment:
Consider implementing a service layer from which you would call the Repository layer like you did it in all other cases.